### DISCUSSION TITLE
Passing parameters to `server/api` or `serverMiddlewares`

### DISCUSSION THREAD

**Discussion by Rigo-m** [nuxt/nuxt#17749] [Ideas]
AFAIK, both `server/api` and `serverMiddlewares` gets exposed as lamba functions in deployment environments (e.g. netlify) but gets called directly by nitro when in SSR/generating (whenever generating will be implemented).

Yes, you can pass POST parameters in request body. But couldn't there be a more elegant (and possibly easier to type) solution?

**Comment by Rigo-m** [comment]
I'll unravel further:

Let's say I have a module for my webservice: `PokémonNuxtModule`, and I want to expose a REST API via adding serverMiddlewares.
Something like this:
```ts
// getPokemonStats.ts
import type { IncomingMessage, ServerResponse } from "http";
import { fetch } from 'oh-my-fetch'
import { useBody, defineHandle } from "h3";

export default defineHandle(
  async (req: IncomingMessage, res: ServerResponse) => {
    const { pokemonId,  getShiny, level } = await useBody(req)
    // use body params to do stuff and return whatever
  }
)
```
When I call with useFetch
```ts
const { data, error, pending } = useFetch('/api/getPokemonStats', { body: { pokemonId: 1, getShiny: false, level: 99 }, method: "POST" })
```

How to assure that the userland passes the correct parameters to the body?
Responses from `/server/api` and `serverMiddlewares` are magically typed, while parameters in request are not. How should we correctly solve this issue?

**Comment by DevJoghurt** [comment]
That is a very good point. I also have a usecase for my nuxt 3 modules where I want to pass a configuration parameter to the server middleware via config. In my case it is the path of the cookie that should be configurable.

**Reply by Rigo-m** [reply]
You can do that already by leveraging runtimeConfig, I think.
See: [Runtime Config](https://v3.nuxtjs.org/guide/features/runtime-config)

**Comment by danielroe** [comment]
See https://github.com/nuxt/framework/issues/573 - Nuxt Functions are coming! 🔥

**Reply by Rigo-m** [reply]
😍 any (approximate) ETA?

**Reply by pi0** [reply]
Hopefully in May. Depends how much pressre nitropack project gets after RC release. But definitely something missing to add!
